src/v/cluster/topics_frontend.cc

```diff
@@ -1045,13 +1133,22 @@ topics_frontend::increase_replication_factor(
         co_return errc::topic_not_exists;
     }
 
+    auto healt_report = co_await get_health_info(
+      topic, tp_metadata->get_configuration().partition_count);
+
     for (const auto& assignment : tp_metadata->get_assignments()) {
         if (assignment.replicas.size() > new_replication_factor) {
             co_return errc::topic_invalid_replication_factor;
         }
 
-        auto partition_constraints = cluster::partition_constraints(
-          assignment.id, new_replication_factor);
+        auto hard_max_disk_usage_ratio = (100 - _hard_max_disk_usage_ratio())
```

Comment:
The compiler will probably optimize this anyway, but it looks like `hard_max_disk_usage_ratio` doesn't change between iterations, so should be calculated once outside the loop